common/static/js/es2015/ReactRenderer.jsx

```diff
@@ -0,0 +1,105 @@
+import 'babel-polyfill';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+
+class ReactRendererException extends Error {
+  constructor(message) {
+    super(`ReactRendererException: ${message}`);
+    Error.captureStackTrace(this, ReactRendererException);
+  }
+}
+
+export class ReactRenderer {
+  constructor({ component, id, componentName, props = {}, store }) {
+    Object.assign(this, {
+      component,
+      id,
+      componentName,
+      props,
+      store,
+    });
+    this.handleArgumentErrors();
+    document.addEventListener('DOMContentLoaded', this.renderComponent.bind(this));
```

Comment:
This is a new addition. Wait for the whole DOM to render (in case consumer is trying to render into an existing DOM element lower down in the DOM)